docs/api.md

```diff
@@ -170,7 +170,6 @@
 - hasSchemaFormat(): `boolean`
 - schemaFormat(): `string` | `undefined`
 - hasMessageId(): `boolean`
```

Comment:
Why is hasMessageId not removed? 🤔

Reply:
Because it makes sense for v2. This API should support all versions so far.

Reply:
Then why is messageId function removed? I dont think you can have one and not the other? 

hasMessageid directly tells you whether you get any information from calling messageId. But when that is removed, what does it tell then?

Reply:
Because we already have `.id()`. If you want to get the id of the message you use `.id()` and if you want to know if this message has the `messageId` property set, then you use `.hasMessageId()`.

In v2, `.id()` would give you the value of `messageId` if it's set and a "calculated one" if it's not. I still want to have a function that tells me if that id is coming from the `messageId` property or if it's "calculated".

Reply:
I think it's a dangerous approach, cause now the user ned to know the underlying code of id, that it prefer messageId over anything else in v2 🤨 

Would it not be better for the UX to keep all the functions and just return undefined for messageId for v3? 🤔

Reply:
Precisely for the UX I'd remove `.messageId()`. Because, without looking at the implementation, what's the difference between `.id()` and `.messageId()`?

If we remove `.messageId()`, then `.id()` becomes the only way to retrieve the id of the message, which makes sense. And only those who are interested in checking for the `messageId` property will use `.hasMessageId()`.

Reply:
`.hasMessageId()` is, in this case, no different than `.hasDescription()`. It's just checking for a property existence.